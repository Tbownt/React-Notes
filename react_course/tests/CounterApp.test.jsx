const { render, screen, fireEvent } = require("@testing-library/react");
const { default: CounterApp } = require("../src/CounterApp");

describe("Pruebas en CounterApp", () => {
  const valor = 100;

  test("debe hacer de hacer match con el snapshot", () => {
    const { container } = render(<CounterApp value={valor} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el valor inicial de 100", () => {
    render(<CounterApp value={valor} />);
    expect(screen.getByText(valor)).toBeTruthy();
  });

  test("debe de incrementar con el boton +1", () => {
    render(<CounterApp value={valor} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText("101")).toBeTruthy();
  });

  test("debe de decrementar con el boton -1", () => {
    render(<CounterApp value={valor} />);
    fireEvent.click(screen.getByText("-1"));
    expect(screen.getByText("99")).toBeTruthy();
  });

  test("debe de resetear el valor del state con el boton de reset", () => {
    render(<CounterApp value={valor} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    // fireEvent.click(screen.getByText("Reset"));
    fireEvent.click(screen.getByRole("button", { name: "btn-reset" }));
    expect(screen.getByText(valor)).toBeTruthy();
    // expect(screen.getByText(valor)).toBeTruthy();
  });
});
