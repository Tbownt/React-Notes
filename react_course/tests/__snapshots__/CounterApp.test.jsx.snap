// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp debe hacer de hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <h3>
      100
    </h3>
    <br />
    <button>
       +1 
    </button>
    <button>
       -1 
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
